Sources/NIO/ChannelInvoker.swift

```diff
@@ -138,49 +138,49 @@ public protocol ChannelOutboundInvoker {
 ///     - return `EventLoopPromise.futureResult`
 extension ChannelOutboundInvoker {
     public func register() -> EventLoopFuture<Void> {
-        let promise = newPromise()
+        let promise: EventLoopPromise<Void> = newPromise()
         register(promise: promise)
         return promise.futureResult
     }
 
     public func bind(to address: SocketAddress) -> EventLoopFuture<Void> {
-        let promise = newPromise()
+        let promise: EventLoopPromise<Void> = newPromise()
         bind(to: address, promise: promise)
         return promise.futureResult
     }
 
     public func connect(to address: SocketAddress) -> EventLoopFuture<Void> {
-        let promise = newPromise()
+        let promise: EventLoopPromise<Void> = newPromise()
         connect(to: address, promise: promise)
         return promise.futureResult
     }
 
     public func write(_ data: NIOAny) -> EventLoopFuture<Void> {
-        let promise = newPromise()
+        let promise: EventLoopPromise<Void> = newPromise()
         write(data, promise: promise)
         return promise.futureResult
     }
 
     public func writeAndFlush(_ data: NIOAny) -> EventLoopFuture<Void> {
-        let promise = newPromise()
+        let promise: EventLoopPromise<Void> = newPromise()
         writeAndFlush(data, promise: promise)
         return promise.futureResult
     }
 
     public func close(mode: CloseMode = .all) -> EventLoopFuture<Void> {
-        let promise = newPromise()
+        let promise: EventLoopPromise<Void> = newPromise()
         close(mode: mode, promise: promise)
         return promise.futureResult
     }
 
     public func triggerUserOutboundEvent(_ event: Any) -> EventLoopFuture<Void> {
-        let promise = newPromise()
+        let promise: EventLoopPromise<Void> = newPromise()
         triggerUserOutboundEvent(event, promise: promise)
         return promise.futureResult
     }
 
-    private func newPromise() -> EventLoopPromise<Void> {
-        return eventLoop.newPromise()
+    public func newPromise<T>(file: StaticString = #file, line: UInt = #line) -> EventLoopPromise<T> {
```

Comment:
@tib can we keep this private for now ? Also I think we can just keep the old return type to remove some typing ;)

Reply:
Sure, I'll make the changes.